internal/mptrie/mptrie.go

```diff
@@ -454,31 +454,36 @@ func (t *MPTrie) Commit(blockNum uint64) error {
 	t.lock.RLock()
 	defer t.lock.RUnlock()
 
-	if err := t.persistSubtrie(t.root); err != nil {
+	rootHash, err := t.root.hash()
+	if err != nil {
+		return err
+	}
+
+	if err := t.persistSubtrie(rootHash); err != nil {
 		return err
 	}
 	return t.store.CommitChanges(blockNum)
 }
 
-func (t *MPTrie) persistSubtrie(n TrieNode) error {
-	wasChanged, err := t.persistNode(n)
+func (t *MPTrie) persistSubtrie(nodePtr []byte) error {
+	wasChanged, err := t.persistNode(nodePtr)
 	if err != nil {
 		return err
 	}
+
 	// If node wasn't changed, no need to persist iт child nodes and can return
 	if !wasChanged {
 		return nil
 	}
-	switch n := n.(type) {
+	node, err := t.store.GetNode(nodePtr)
+	if err != nil {
+		return err
+	}
+	switch n := node.(type) {
 	case *BranchNode:
 		for _, childPtr := range n.Children {
 			if childPtr != nil {
-				child, err := t.store.GetNode(childPtr)
-				if err != nil {
-					return err
-				}
-
-				if err = t.persistSubtrie(child); err != nil {
+				if err = t.persistSubtrie(childPtr); err != nil {
```

Comment:
While you are already at it, can you please reverse the logic of if statement above to reduce code indentation?

Reply:
Done, but it adds extra code line :)

Reply:
Done